frontend/model/contracts/group.js

```diff
@@ -259,6 +261,29 @@ DefineContract({
         }
       }
     },
+    'gi.contracts/group/updateSettings': {
+      // OPTIMIZE: Make this custom validation function
+      // reusable accross other future validators
+      validate: data => {
+        const validations = {
+          mincomeAmount: x => typeof x === 'number' && x > 0,
+          mincomeCurrency: x => typeof x === 'string'
+        }
+        for (const key in data) {
+          if (validations[key]) {
+            if (!validations[key](data[key])) {
+              throw new Error(`${key} has bad value: ${data[key]}`)
```

Comment:
Once #694 is closed this won't be an issue, but for now this needs to be `TypeError` instead of `Error`

Reply:
done